### ISSUE TITLE
How much is the final training loss?

### ISSUE THREAD

**Issue by Todd-Qi** [JiayuYANG/CVP-MVSNet#3]
Thanks for your amazing work. @JiayuYANG @MiaomiaoLiu 
1. The 3D points(DTU) produced by your **pretrained** `model_0027.ckpt` is consistent with your paper.
2. I have re-trained the network with bactch_size=12. In this case, the 3D points(DTU) produced by **my** `model_0027.ckpt` is worse than the result in your paper, using the same depth fusion method. 
3. Besides, the traing loss of **my** `model_0027.ckpt` is ~52. I think it's a bit big. So I wonder how much is your final training loss?

Could you please provide a link of your Supp. Material if you are convenient? 

Looking forward to you reply.
Best Regards.

**Comment by JiayuYANG** [comment]
Hi Todd-Qi,

Thank you for your interest in our project.

Regarding your questions:
The final training loss of our pre-trained model is also around 50, so your final loss is reasonable.
However we used batch size 16 on training. Smaller batch size may cause more noise during training.
And about your result, what numbers have you got on DTU dataset?
Could you please provide the evaluation parameters and fusion parameters you used to get the result?

Cheers,
Jiayu

**Comment by XYZ-qiyh** [comment]
Hello, @JiayuYANG 
I'm using the same fusion parameter, i.e. `prob_threshold=0.8`, `disp_threshold=0.13` and `num_consistent=3`.
I think the problem comes from the predicted depth map. The depth map of mine has many holes.
![rect_007_3_r5000](https://user-images.githubusercontent.com/45621782/80094491-e911c300-8598-11ea-96cf-b0a7c3d7739a.png)
![00000007 pfm](https://user-images.githubusercontent.com/45621782/80094496-eb741d00-8598-11ea-87f4-cd013928dbef.png)
![00000007 pretrain](https://user-images.githubusercontent.com/45621782/80094500-ed3de080-8598-11ea-8c9c-3aba9392ccf5.png)
from the top to bottom, (a)the original ref image, (b)the depth map predicted by my trained model, (c)the depth map predicted by your pretrained model.
For the evaluation, the number of depth sampling is **48**.

In addition, I compute the mean absolute error(MAE) on the DTU validtion set. The MAE of your pretrained model is about 10 while my trained model is about 15.

Do you have any suggestions how to solve this problem?
Thank You！

**Comment by JiayuYANG** [comment]
Hi Todd-Qi,

I have updated our code to solve this issue,
you may pull the latest version to reproduce our results.

Current version produced following results when training from scratch using the default settings.
| acc.   | comp.  | overall |
|--------|--------|---------|
| 0.3032 | 0.4161 | 0.3597  |

Cheers,
Jiayu

**Comment by etudemin** [comment]
Hi @JiayuYANG,

Thanks for your great work! 

I've retrained your network using the default setting, and got 40 models which correspond to 40 epochs.

In this project, authors provided the pre-trained model of the 27th epoch (model_0027.ckpt). Would you kindly explain how the authors choose the specific model?

Attached is the generated log file, and the epoch loss is dropping gradually during 40 epochs. Seems that there is no validation process during training, so it's a little confusing. Thanks for your kind reply in advance, and please correct me if I miss something. Thanks again! 
[20200831-0132.log](https://github.com/JiayuYANG/CVP-MVSNet/files/5179263/20200831-0132.log)
